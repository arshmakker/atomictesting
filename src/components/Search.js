import React from 'react';
import Button from './Button';


function Search({
  classSearch= "",
  title="Search",
  searchPlaceholder="Type here to search",
  onSearch=''
} ) {
  
  return (
    <div className={classSearch}>
      <Button title={title} onClick={onSearch}></Button>
      <input type="text" id="searchText" placeholder={searchPlaceholder}/>
    </div>
  )
}

export default Search;