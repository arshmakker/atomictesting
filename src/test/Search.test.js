import React from 'react';
import Search from '../components/Search'
import renderer from 'react-test-renderer'

const classSearch = "sampleClass"
const title = "Sample Title"
const searchPlaceholder = "Sample PlaceHolder"
const onSearch = ()=>{
    alert("Alert test")
}

it('Search component is rendered correctly with no props',()=>{
    const tree = renderer
    .create(<Search />)
    .toJSON();
expect(tree).toMatchSnapshot();
})
it('Search component is rendered correctly with test props',()=>{
   
    const tree = renderer
    .create(<Search classSearch={classSearch}
        searchPlaceholder={searchPlaceholder}
        title={title}
        onSearch={onSearch} ></Search>)
    .toJSON();
expect(tree).toMatchSnapshot();
})