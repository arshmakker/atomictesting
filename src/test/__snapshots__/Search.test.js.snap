// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component is rendered correctly with no props 1`] = `
<div
  className=""
>
  <button
    className=""
  >
    Search
  </button>
  <input
    id="searchText"
    placeholder="Type here to search"
    type="text"
  />
</div>
`;

exports[`Search component is rendered correctly with test props 1`] = `
<div
  className="sampleClass"
>
  <button
    className=""
  >
    Sample Title
  </button>
  <input
    id="searchText"
    placeholder="Sample PlaceHolder"
    type="text"
  />
</div>
`;
